cts/cli/regression.rules.exp

```diff
@@ -28,6 +28,7 @@ crm_rule: Couldn't parse input from STDIN
 * Passed: echo           - crm_rule given invalid input XML on stdin
 =#=#=#= Begin test: Try to check a rule that doesn't exist =#=#=#=
 No rule found with ID=blahblah
+Could not determine whether rule blahblah is expired
```

Comment:
@clumens Example of new text output when we call `out->message` in the event of rule-check errors too.

Reply:
The text seems redundant, not sure what we should do here. The ideal would be something like

    Could not determine whether rule is expired: Rule blahblah not found

but that's probably not worth the hassle of figuring out

Reply:
I think it can be done without too much trouble, if we want to send the "Could not determine" message to `out->err` instead of `out->info` like we do with the other formatter messages.

Reply:
Seems like it should be an error anyway

Reply:
@kgaillot I've got it working so that it all prints on one line for "rule not found" and "more than one rule" cases for text output. The unsupported date format case would still have two lines, because we'd be returning `EOPNOTSUPP` for both subcases and the formatter has no way to differentiate.

The only caveat is that the error **messages** for "rule not found" and "more than one rule" won't show up in XML output. XML output would only get the return code. (XML output would still get the error messages for the unsupported date formats.)

That might be okay, as long as we use each return code for a unique situation: `ENXIO` for "rule not found" and `pcmk_rc_duplicate_id` for "more than one rule." I'm doing that in the next update regardless.

Thoughts?

```
Could not determine whether rule blahblah is expired: Rule not found
...
Could not determine whether rule duplicate-rule-id is expired: Found more than one rule with matching ID
...
Rule cli-prefer-rule-dummy-date_spec-years-moon must either not use date_spec, or use date_spec with years= but not moon=
Could not determine whether rule cli-prefer-rule-dummy-date_spec-years-moon is expired: Unsupported date format
...
Can't check rule cli-no-date_expression-rule because it does not have exactly one date_expression
Could not determine whether rule cli-no-date_expression-rule is expired: Unsupported date format
```

```
<pacemaker-result api-version="X" request="crm_rule -c -r blahblah --output-as=xml">
  <rule-check rule-id="blahblah" rc="105"/>
  <status code="105" message="No such object">
  <!-- this would go away
    <errors>
      <error>No rule found with ID=blahblah</error>
    </errors>
  -->
  </status>
</pacemaker-result>
```

Reply:
I'll go ahead and push it with the other changes. It's easy to modify if we decide we don't want this.